pkg/odo/util/pushtarget/pushtarget_test.go

```diff
@@ -0,0 +1,114 @@
+package pushtarget
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/openshift/odo/pkg/preference"
+)
+
+func TestIsDockerPushTargetEnabled(t *testing.T) {
+
+	const (
+		pushTargetSetting string = "PushTarget"
+	)
+
+	// temp config file
+	tempConfigFile, err := ioutil.TempFile("", "odoconfig")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tempConfigFile.Close()
+
+	// set global config env
+	os.Setenv(preference.GlobalConfigEnvName, tempConfigFile.Name())
+
+	// test table
+	tests := []struct {
+		name      string
+		setEnv    string
+		setConfig string
+		want      bool
+	}{
+		{
+			name:      "no pushtarget setting set",
+			setEnv:    "",
+			setConfig: "",
+			want:      false,
+		},
+		{
+			name:      "set pushtarget to docker in config",
+			setEnv:    "",
+			setConfig: "docker",
+			want:      true,
+		},
+		{
+			name:      "set pushtarget to kube in config",
+			setEnv:    "",
+			setConfig: "kube",
+			want:      false,
+		},
+		{
+			name:      "enable docker pushtarget in env",
+			setEnv:    "docker",
+			setConfig: "",
+			want:      true,
+		},
+		{
+			name:      "set pushtarget to kube in env",
+			setEnv:    "kube",
+			setConfig: "",
+			want:      false,
+		},
+		{
+			name:      "override pushtarget prefrence with docker pushtarget env",
+			setEnv:    "docker",
+			setConfig: "kube",
+			want:      true,
+		},
+		{
+			name:      "override pushtarget prefrence with kube pushtarget env",
+			setEnv:    "kube",
+			setConfig: "docker",
+			want:      false,
+		},
+	}
```

Comment:
please add a case where the env is neither `kube` nor `docker`

Reply:
Done, updated!